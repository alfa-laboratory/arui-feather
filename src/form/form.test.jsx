/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from 'react';
import { shallow, mount } from 'enzyme';

import Form from './form';

describe('form', () => {
    it('should render without problems', () => {
        let form = shallow(<Form>Form-example</Form>);

        expect(form).toMatchSnapshot();
        expect(form.text()).toContain('Form-example');
    });

    it('should render footer at the end of a form', () => {
        let form = shallow(<Form footer={ <div>footer</div> }>Form-content</Form>);
        let contentNode = form.find('.form__footer');

        expect(contentNode.text()).toContain('footer');
    });

    it('should call `onSubmit` callback after form was submitted', () => {
        let onSubmit = jest.fn();
        let form = mount(<Form onSubmit={ onSubmit } />);

        form.simulate('submit');

        expect(onSubmit).toHaveBeenCalled();
    });
});
