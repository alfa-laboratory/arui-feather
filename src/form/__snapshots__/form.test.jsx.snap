// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form should render without problems 1`] = `
<form
  action="/"
  autoComplete="on"
  className="form form_size_m form_theme_alfa-on-white"
  encType="application/x-www-form-urlencoded"
  method="post"
  noValidate={false}
  onSubmit={[Function]}
>
  Form-example
</form>
`;
